Port IPC::Semaphore to the new IPC serialization format

cdumez opened the pull request:
#### bdc2cb3920a7260e347d3a7b01fd116959761bbc
<pre>
Port IPC::Semaphore to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=264911">https://bugs.webkit.org/show_bug.cgi?id=264911</a>

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/IPC/IPCSemaphore.h:
* Source/WebKit/Platform/IPC/IPCSemaphore.serialization.in: Added.
* Source/WebKit/Platform/IPC/darwin/IPCSemaphoreDarwin.cpp:
(IPC::Semaphore::encode const): Deleted.
(IPC::Semaphore::decode): Deleted.
* Source/WebKit/Platform/IPC/unix/IPCSemaphoreUnix.cpp:
(IPC::Semaphore::duplicateDescriptor const):
(IPC::Semaphore::encode const): Deleted.
(IPC::Semaphore::decode): Deleted.
* Source/WebKit/Platform/IPC/win/IPCSemaphoreWin.cpp:
(IPC::Semaphore::toWin32Handle const):
(IPC::Semaphore::encode const): Deleted.
(IPC::Semaphore::decode): Deleted.
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPCSemaphore::encode const):

Canonical link: <a href="https://commits.webkit.org/270845@main">https://commits.webkit.org/270845@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e7180cee2f03fcf641a1f178f05abedcf39d6555

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26593 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5208 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27840 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28805 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24320 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7021 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2615 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/24269 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26854 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4041 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22845 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3555 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/3599 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29290 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24255 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24231 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29864 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3632 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1834 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27761 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5070 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6385 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4106 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3963 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Platform/IPC/IPCSemaphore.h:
```diff
@@ -76,8 +73,11 @@ class Semaphore {
     explicit operator bool() const { return m_sendRight || m_semaphore != SEMAPHORE_NULL; }
 #elif OS(WINDOWS)
     explicit Semaphore(Win32Handle&&);
+    Win32Handle toWin32Handle();
```
I'm not sure "to" is a good description of what this accessor is doing.  Also, this could probably be inline in the header.

cdumez replied:
Ok, I'm still fighting to get this particular getter to build with the wincairo EWS. I'll refine.

cdumez committed:
Port IPC::Semaphore to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=264911

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/IPC/IPCSemaphore.h:
* Source/WebKit/Platform/IPC/IPCSemaphore.serialization.in: Added.
* Source/WebKit/Platform/IPC/darwin/IPCSemaphoreDarwin.cpp:
(IPC::Semaphore::encode const): Deleted.
(IPC::Semaphore::decode): Deleted.
* Source/WebKit/Platform/IPC/unix/IPCSemaphoreUnix.cpp:
(IPC::Semaphore::duplicateDescriptor const):
(IPC::Semaphore::encode const): Deleted.
(IPC::Semaphore::decode): Deleted.
* Source/WebKit/Platform/IPC/win/IPCSemaphoreWin.cpp:
(IPC::Semaphore::toWin32Handle const):
(IPC::Semaphore::encode const): Deleted.
(IPC::Semaphore::decode): Deleted.
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPCSemaphore::encode const):

Canonical link: https://commits.webkit.org/270845@main
